rust/parser/mod.rs

```diff
@@ -305,15 +309,111 @@ fn visit_value_type_primitive(node: Node<'_>) -> BuiltinValueType {
 
 fn visit_value_literal(node: Node<'_>) -> Literal {
     debug_assert_eq!(node.as_rule(), Rule::value_literal);
-    Literal::new(node.span(), None, node.as_str().to_owned()) // TODO visit to get category
+    let span = node.span();
+    let child = node.into_child();
+    let value_literal = match child.as_rule() {
+        Rule::quoted_string_literal => ValueLiteral::String(visit_quoted_string_literal(child)),
+        Rule::boolean_literal => ValueLiteral::Boolean(BooleanLiteral { value: child.as_str().to_owned() }),
+        Rule::signed_integer => ValueLiteral::Integer(visit_signed_integer(child)),
+        Rule::signed_decimal => ValueLiteral::Decimal(visit_signed_decimal(child)),
+
+        Rule::datetime_tz_literal => ValueLiteral::DateTimeTz(visit_datetime_tz_literal(child)),
+        Rule::datetime_literal => ValueLiteral::DateTime(visit_datetime_literal(child)),
+        Rule::date_literal => ValueLiteral::Date(visit_date_literal(child)),
+
+        _ => unreachable!("{}", TypeQLError::IllegalGrammar { input: child.to_string() }),
+    };
+    Literal::new(span, value_literal)
 }
 
-fn visit_quoted_string_literal(node: Node<'_>) -> Literal {
-    debug_assert_eq!(node.as_rule(), Rule::quoted_string_literal);
-    Literal::new(node.span(), Some(Tag::String), node.as_str().to_owned())
+fn visit_sign(node: Node<'_>) -> Sign {
+    debug_assert_eq!(node.as_rule(), Rule::sign);
+    match node.as_rule() {
+        Rule::PLUS => Sign::Plus,
+        Rule::MIN => Sign::Minus,
```

Comment:
This is wrong. Run `cargo test -- autogenerated`, it is meant to catch typos like these

Reply:
Ah, this didn't run through the IDE. I guess that's why I didn't catch them.